peer/abstractlist/list.go

```diff
@@ -548,31 +557,38 @@ func (pl *List) newNoContextDeadlineError() error {
 }
 
 func (pl *List) newUnavailableError(err error) error {
-	return yarpcerrors.Newf(yarpcerrors.CodeUnavailable, "%q peer list timed out waiting for peer: %s", pl.name, err.Error())
+	return yarpcerrors.Newf(yarpcerrors.CodeUnavailable, "%q peer list timed out waiting for peer, %s assigned, fail-fast is not enabled: %s", pl.name, pl.formatNumPeers(), err.Error())
```

Comment:
- Could we indicate the availability of peer we're waiting for? ie `... timed out waiting for an available peer`
- and instead of `assigned` what if we used `unavailable`

This would read like:
```
"%q peer list timed out waiting for an available peer (%s unavailable), fail-fast is disabled: %s"
```